WrightTools/kit/_file.py

```diff
@@ -0,0 +1,101 @@
+"""Tools for interacting with files."""
+
+
+# --- import --------------------------------------------------------------------------------------
+
+
+import linecache
+
+
+# --- define --------------------------------------------------------------------------------------
+
+
+__all__ = ['file_len', 'FileSlicer']
+
+
+# --- functions -----------------------------------------------------------------------------------
+
+
+def file_len(fname):
+    """Get the number of lines in a file, without loading into memory."""
+    # adapted from http://stackoverflow.com/questions/845058
+    with open(fname) as f:
+        for i, l in enumerate(f):
+            pass
+    return i + 1
+
+
+# --- class ---------------------------------------------------------------------------------------
+
+
+class FileSlicer:
+    """Access groups of lines from a file quickly, without loading the entire file into memory.
+
+    Mostly a convinient wrapper around the standard library linecache
+    module.
+    """
+
+    def __init__(self, path, skip_headers=True, header_charachter='#'):
```

Comment:
No time like the present to fix a spelling mistake: charachter --> character
(I know this is not new, but as it's being moved/we are going to v3.0.0)